lib/workflows/email/send_decom_flavor_email.py

```diff
@@ -0,0 +1,141 @@
+    :param all_projects: A boolean which if selected will search in all projects
+    :param send_email: Actually send the email instead of printing what will be sent
+    :param as_html: Send email as html
+    :param kwargs: see EmailParams dataclass class docstring
+    """
+
+    if project_list and all_projects:
+        raise RuntimeError(
+            "given both project list and all_projects flag - please choose only one"
+        )
+
+    # flavor query
+    q1 = FlavorQuery().where(
+        QueryPresetsGeneric.EQUAL_TO, FlavorProperties.FLAVOR_IS_PUBLIC, value=True
+    )
+    q1.where(
+        QueryPresetsGeneric.ANY_IN,
+        FlavorProperties.FLAVOR_NAME,
+        values=decom_flavor_list,
+    )
+    q1.run(cloud_account)
+    q1.sort_by((FlavorProperties.FLAVOR_ID, False))
+
+    # server query
+    q2 = q1.then(query_type="SERVER_QUERY", forward_outputs=True).run(
+        cloud_account, as_admin=True, all_projects=True
+    )
+    q2.append_from("PROJECT_QUERY", cloud_account, ProjectProperties.PROJECT_NAME)
+    q2.select(
+        ServerProperties.SERVER_ID,
+        ServerProperties.SERVER_NAME,
+        ServerProperties.ADDRESSES,
+    )
+
+    # user query
+    q3 = q2.then("USER_QUERY", forward_outputs=True).run(cloud_account)
+    q3.select(UserProperties.USER_NAME)
+    q3.group_by(UserProperties.USER_EMAIL)
+    decom_results = q3.to_list()
+
+    for email_addr, outputs in decom_results.items():
+        user_name = outputs[0]["user_name"]
+        for output in outputs:
+            del output["user_name"]
+
+        flavor_list = [["Flavor Name"]]
+        flavor_list.extend([[flavor] for flavor in decom_flavor_list])
+        flavor_table = tabulate(flavor_list)
+
+        if not send_email:
+            print(f"Send Email To: {email_addr}")
+            print(f"email_templates decom-email: username {user_name}")
+            print(f"send as html: {as_html}")
+            print(f"flavor table: {flavor_table}")
+            print(f"decom table: {q3.to_string(groups=[email_addr])}")
+            print("\n\n")
+            break
+
+        email_params = EmailParams.from_dict(
+            {
+                **{
+                    "email_templates": [
+                        EmailTemplateDetails(
+                            template_name="decom_email",
+                            template_params={
+                                "username": user_name,
+                                "affected_flavors_table": flavor_table,
+                                "decom_table": q3.to_string(groups=[email_addr]),
+                            },
+                        ),
+                        EmailTemplateDetails(
+                            template_name="footer", template_params={}
+                        ),
+                    ],
+                    "email_to": email_addr,
+                    "as_html": as_html,
+                },
+                **kwargs,
```

Comment:
I really don't like this, yes it does it in a single step but it's unweildly to look at
Can we do it as three statements

Also why do we make two different email template details? This feels really weird from an API so I think I'd want to change it to take a single `EmailTemplateDetails` object

Lastly, we should use kwargs such as `email_to` directly and not (ab)use the fact we can decompile a dict into kwargs, it's not very future developer friendly to do this way

Reply:
1. Email is constructed from multiple Templates - the header (decom_email) and footer (footer) 

we _can_ have a define a single dataclass hold multiple Templates - either way we need to define each template separately - as they take different params

2. Yea it's a mess I agree - I'll clean it up

Reply:
Can't we just have a header and footer in `EmailTemplateDetails` - I can't think of a time we'd ever need `template_params` for the footer? So it can just be hardcoded to None/Empty dict

Reply:
Happy to hardcode header and footer - but this doesn't fix your issue

The underlying question to ask here is 
will we ever require creating an email with multiple templates in the body? 

If yes - we'll need to define templates separately like we've done here - and build EmailParams datastructure from multiple templates. 

If no - I'd argue that we should support having multiple templates anyway - since we can make those templates DRY

Reply:
Let's see what it's like with a cleaner call

I'm sure if it looked like
```
body_template = EmailTemplateDetails(...)
footer_template = EmailTemplateDetails(...)

email_api.send_email (
   email_templates = [body_template, footer_template],
   email_to = email_addr
)
```
I wouldn't have noticed / flagged it